src/Castle.Windsor/Compatibility/ApplicationDomains.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Castle.Core.Internal
+{
+    internal static class ApplicationDomains
```

Comment:
Changed ApplicationDomains into AppContext. This file is duplicated into the Castle.Windsor.Tests project, because internals are not visible to tests. I will create a shared project in the solution, with AppContext.cs and NetCoreReflectionExtensions.cs.